text/0101-new-scale-control-model-based-on-IO-size.md

```diff
@@ -0,0 +1,82 @@
+
+## Flow Mechanism
+In order to prevent PD from over-delivering snapshot tasks to the TIKV instance. Each store will have sliding windows to send snapshots, and the total size of the sliding window can be adjusted by the snapshot execution duration and the operator execution. 
+```mermaid
+graph LR
+    subgraph PD
+            subgraph Store
+                RS[regions]
+                SW[sliding window]
+            end
+      SC[Scheduler]-->|1. pick region|RS
+      RS -->|2. check store|SW
+            RS -->|3. region available|SC 
+      SC -->|4. put operator and take token| OC[operator controller]    
+    end 
+
+    
+    
+    subgraph TIKV-leader
+      OC -.->|5. region heartbeat| R[Raft]
+      R -.->|6. generate task| P[pool]
+      P -->|FIFO|P
+      P -.->|7. send task| SE[sender]
+            P -->|7.2 generate duration +|Z
+            SE-->|8.2 send duration +|Z[collect]
+      
+      Z -.->|9.2 store heartbeat|SW
+    end 
+    
+    subgraph TIKV-learner
+        SE -->|8. send snapshot|RE[receiver]
+     RE-->|9. apply snapshot|R1[raft]
+    end
+```
+
+The mechanism of the scheduling limiter is shown in the figure. All scheduler will check the region that's leader has some available send size. If not , it will try another region as target.
+
+### How to adjust windows size 
+The minimum(initial) value of the sliding window is 1000MB. The store can be considered to be scheduled if it's available sending size is bigger than 100MB, so the biggest region can also be scheduled. 
+When the operator execution duration is less than 2 times the snapshot task execution time, the sender windows size will increase , otherwise it will decrease. A PI controller is used to prevent repeated adjustments. It should be noted that when all schedules don't need such big size, it will stop increasing. 
+
+### How to ensure fairness 
+The new scheduler policy does not guarantee that the snapshot sending size of all stores is exactly the same, but will switch to the next candidate region when the store has no available sending size. During the prototype testing process on the scale-out and scale-in scene , the sending size on all TIKV reaches the bottleneck. 
+
+### Scheduler 
+#### Balance Region Scheduler 
+The scheduler picks the store as source first, then picks the region on the source store, finally picks the target store. On the original basis, it is necessary to check the target region leader whether there is enough sending space to send. 
+#### High Priority Scheduler 
+Some scheduler may create a highest priority operator such as balance hot region, region check. For these higher operators, 20% of the send window will be reserved. This can prevent this window size from being completely occupied by low priority task .
```

Comment:
How about adding more details about "20% of the send window will be reserved"?

Reply:
fix